### PR TITLE
re-enabling jwt on keymanager API

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

https://ethereum.github.io/keymanager-APIs

I noticed that the keymanager APIs do have the appropriate codes if using a jwt token, so I am re-enabling the middleware to handle it in the following pr 

Tests 
```
curl -X GET "http://localhost:7500/eth/v1/keystores" -H "accept: application/json"
unauthorized: no Authorization header passed. Please use an Authorization header with the jwt created in the prysm wallet

curl -X GET "http://localhost:7500/eth/v1/keystores" -H "accept: application/json"   -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.vgs336GzjwJZaYV7lrmUS1swwVXQqsA1_TB42-7ewIE"
{"data":[{"validating_pubkey":"0x855ae9c6184d6edd46351b375f16f541b2d33b0ed0da9be4571b13938588aee840ba606a946f0e8023ae3a4b2a43b4d4","derivation_path":""},{"validating_pubkey":"0xaa0ef7404c3a164741856c78cc844072448886291e3c5b29514001d1bd8c5ccc9c342a1caa58c3b45734195ef4b4122c","derivation_path":""}]}
```
Also tested on web UI

### CODE REVIEW

File: validator/rpc/intercepter.go [hunk 1]
```diff
@@ -50,7 +49,8 @@ func (s *Server) JwtHttpInterceptor(next http.Handler) http.Handler {
 			token := strings.Split(reqToken, "Bearer ")[1]
 			_, err := jwt.Parse(token, s.validateJWT)
 			if err != nil {
-				http.Error(w, fmt.Errorf("unauthorized:could not parse JWT token: %v", err).Error(), http.StatusUnauthorized)
+				fmt.Println(err)
+				http.Error(w, fmt.Errorf("unauthorized:could not parse JWT token: %v", err).Error(), http.StatusForbidden)
```
Comment (saolyn) [inline validator/rpc/intercepter.go:53]:
```suggestion
				http.Error(w, fmt.Errorf("forbidden: could not parse JWT token: %v", err).Error(), http.StatusForbidden)
```

File: validator/rpc/intercepter.go [hunk 2]
```diff
@@ -40,8 +40,7 @@ func (s *Server) JwtHttpInterceptor(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// if it's not initialize or has a web prefix
 		if !strings.Contains(r.URL.Path, api.WebUrlPrefix+"initialize") && // ignore some routes
-			!strings.Contains(r.URL.Path, api.WebUrlPrefix+"health/logs") &&
-			strings.Contains(r.URL.Path, api.WebUrlPrefix) {
+			!strings.Contains(r.URL.Path, api.WebUrlPrefix+"health/logs") {
```
Comment (saolyn) [inline validator/rpc/intercepter.go:43]: Do you need the prefix? Could you not just check:
```
if !strings.Contains(r.URL.Path, "initialize") && // ignore some routes
			!strings.Contains(r.URL.Path, "health/logs") {
```
Is there more than one expected endpoint? If not we could just check that one?

Comment (saolyn) [reply validator/rpc/intercepter.go:43]: Sorry looks like my browser was displaying things weirdly, this comment is irrelevant

Comment (james-prysm) [reply validator/rpc/intercepter.go:43]: yeah i think it'll be better with the prefix to indicate it, I'm afraid of other health endpoints down the line if we change that.

File: validator/rpc/intercepter.go [hunk 3]
```diff
@@ -50,7 +49,8 @@ func (s *Server) JwtHttpInterceptor(next http.Handler) http.Handler {
 			token := strings.Split(reqToken, "Bearer ")[1]
 			_, err := jwt.Parse(token, s.validateJWT)
 			if err != nil {
-				http.Error(w, fmt.Errorf("unauthorized:could not parse JWT token: %v", err).Error(), http.StatusUnauthorized)
+				fmt.Println(err)
```
Comment (saolyn) [inline validator/rpc/intercepter.go:52]: Debug log?

File: validator/rpc/intercepter.go [hunk 4]
```diff
@@ -84,5 +84,6 @@ func (s *Server) validateJWT(token *jwt.Token) (interface{}, error) {
 	if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
 		return nil, fmt.Errorf("unexpected JWT signing method: %v", token.Header["alg"])
 	}
+	fmt.Println(s.jwtSecret)
```
Comment (saolyn) [inline validator/rpc/intercepter.go:87]: Debug log?

Comment (james-prysm) [reply validator/rpc/intercepter.go:87]: oops yes will remove

**Review by saolyn — APPROVED** [review]

**Pull request merged**